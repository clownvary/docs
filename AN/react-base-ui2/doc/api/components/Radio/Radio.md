## Radio 
---
UI component of Radio

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|string &#124; number|&ensp;|false|Radio value.
disabled|bool|&ensp;|false|Determines enable/disable state.
size|string|&ensp;|false|Determines the Radio size.
name|string|&ensp;|false|Field name.
children|node|&ensp;|false|Child nodes.
onChange|func|identity|false|Fires when value change.