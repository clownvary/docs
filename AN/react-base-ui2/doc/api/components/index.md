## CollapsePanel 
---
UI component of CollapsePanel

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
children|node|&ensp;|true|child content of CollapsePanel
className|string|''|false|customize class for CollapsePanel section
expandedIcon|string|'icon-chevron-up'|false|expanded icon of CollapsePanel panel
collapsedIcon|string|'icon-chevron-down'|false|collapsed icon of CollapsePanel panel
transition|string|'none'|false|define collapse/expanded transition, just have a transition on attribute of 'height'
title|string|''|false|define what title will be display on collapse panel
summary|string &#124; element|''|false|define what summary will be display on collapse panel
onExpand|func|() => {}|false|a callback function after expanded
onCollapse|func|() => {}|false|a callback function after collapsed