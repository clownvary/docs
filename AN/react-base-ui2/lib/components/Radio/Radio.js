'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _extends2 = require('babel-runtime/helpers/extends');var _extends3 = _interopRequireDefault(_extends2);var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);var _defineProperties = require('babel-runtime/core-js/object/define-properties');var _defineProperties2 = _interopRequireDefault(_defineProperties);var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');
var _identity = require('lodash/identity');var _identity2 = _interopRequireDefault(_identity);

var _RadioComponent = require('./RadioComponent');var _RadioComponent2 = _interopRequireDefault(_RadioComponent);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

/** Default PropTypes of Radio.
                                                                                                                                                                                                                * @memberof Radio
                                                                                                                                                                                                               */
var RadioPropTypes = {
  /** Radio value.
                        * @type {string|number}
                        */
  value: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.number]),
  /** Determines enable/disable state.
                                                                             * @type {boolean}
                                                                             */
  disabled: _propTypes.bool,
  /** Determines the Radio size.
                              * @type {Size}
                              */
  size: _propTypes.string,
  /** Field name.
                            * @type {string}
                            */
  name: _propTypes.string,
  /** Child nodes.
                            * @type {node}
                            */
  children: _propTypes.node,
  /**  Fires when value change.
                              * @event
                              * @type {func}
                             */
  onChange: _propTypes.func };


var RadioProps = {
  onChange: _identity2.default };


/** UI component of Radio */var
Radio = function (_Component) {(0, _inherits3.default)(Radio, _Component);function Radio() {var _ref;var _temp, _this, _ret;(0, _classCallCheck3.default)(this, Radio);for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = Radio.__proto__ || (0, _getPrototypeOf2.default)(Radio)).call.apply(_ref, [this].concat(args))), _this), _this.
























    setRef = function (radioComponent) {
      _this.radioComponent = radioComponent;
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);}(0, _createClass3.default)(Radio, [{ key: 'componentDidMount', // Support legacy API for `this.refs.checkbox.checked`
    // It's not recommended because it breaks component encapsulation and
    // it violates the unidirectional data flow for React
    value: function componentDidMount() {(0, _defineProperties2.default)(this, { checked: { get: function get() {return this.radioComponent.checked;}, set: function set(v) {this.radioComponent.checked = !!v;} } });} }, { key: 'render', value: function render() {var _props = this.props,children = _props.children,rest = (0, _objectWithoutProperties3.default)(_props, ['children']);var
      auiRadioGroup = this.context.auiRadioGroup;

      // Proxy `onChange`, `checked` and `disabled` to `auiRadioGroup`
      if (auiRadioGroup) {
        rest.onChange = auiRadioGroup.onChange;
        rest.checked = this.props.value === auiRadioGroup.value;
        rest.disabled = this.props.disabled || auiRadioGroup.disabled;
        rest.size = this.props.size || auiRadioGroup.size;
        rest.name = this.props.name || auiRadioGroup.name;
      }

      return (
        _react2.default.createElement(_RadioComponent2.default, (0, _extends3.default)({ ref: this.setRef }, rest),
          children));


    } }]);return Radio;}(_react.Component);Radio.displayName = 'Radio';Radio.propTypes = RadioPropTypes;Radio.defaultProps = RadioProps;Radio.contextTypes = { auiRadioGroup: _propTypes.object };exports.default =


Radio;module.exports = exports['default'];