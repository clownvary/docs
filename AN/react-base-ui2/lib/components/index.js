'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _Breadcrumb = require('./Breadcrumb/');var _Breadcrumb2 = _interopRequireDefault(_Breadcrumb);
var _Calendar = require('./Calendar/');var _Calendar2 = _interopRequireDefault(_Calendar);
var _InputMask = require('./InputMask');var _InputMask2 = _interopRequireDefault(_InputMask);
var _InputMoment = require('./InputMoment');var _InputMoment2 = _interopRequireDefault(_InputMoment);
var _InputDate = require('./InputDate');var _InputDate2 = _interopRequireDefault(_InputDate);
var _InputTime = require('./InputTime');var _InputTime2 = _interopRequireDefault(_InputTime);
var _InputTimeRange = require('./InputTimeRange');var _InputTimeRange2 = _interopRequireDefault(_InputTimeRange);
var _InputDateTime = require('./InputDateTime');var _InputDateTime2 = _interopRequireDefault(_InputDateTime);
var _InputNumeric = require('./InputNumeric');var _InputNumeric2 = _interopRequireDefault(_InputNumeric);
var _List = require('./List');var _List2 = _interopRequireDefault(_List);
var _DatePicker = require('./DatePicker');var _DatePicker2 = _interopRequireDefault(_DatePicker);
var _CollapsePanel = require('./CollapsePanel');var _CollapsePanel2 = _interopRequireDefault(_CollapsePanel);
var _LoadingBar = require('./LoadingBar');var _LoadingBar2 = _interopRequireDefault(_LoadingBar);
var _PropertyList = require('./PropertyList');var _PropertyList2 = _interopRequireDefault(_PropertyList);
var _ComboBox = require('./ComboBox');var _ComboBox2 = _interopRequireDefault(_ComboBox);
var _Checkbox = require('./Checkbox');var _Checkbox2 = _interopRequireDefault(_Checkbox);
var _Radio = require('./Radio');var _Radio2 = _interopRequireDefault(_Radio);
var _Input = require('./Input');var _Input2 = _interopRequireDefault(_Input);
var _Button = require('./Button');var _Button2 = _interopRequireDefault(_Button);
var _Modal = require('./Modal');var _Modal2 = _interopRequireDefault(_Modal);
var _Popover = require('./Popover');var _Popover2 = _interopRequireDefault(_Popover);
var _TextArea = require('./TextArea');var _TextArea2 = _interopRequireDefault(_TextArea);
var _AsyncContent = require('./AsyncContent');var _AsyncContent2 = _interopRequireDefault(_AsyncContent);
var _Progress = require('./Progress');var _Progress2 = _interopRequireDefault(_Progress);
var _FileUpload = require('./FileUpload');var _FileUpload2 = _interopRequireDefault(_FileUpload);
var _Survey = require('./Survey');var _Survey2 = _interopRequireDefault(_Survey);
var _ButtonBar = require('./ButtonBar');var _ButtonBar2 = _interopRequireDefault(_ButtonBar);
var _Tabs = require('./Tabs');var _Tabs2 = _interopRequireDefault(_Tabs);
var _Select = require('./Select');var _Select2 = _interopRequireDefault(_Select);
var _Tag = require('./Tag');var _Tag2 = _interopRequireDefault(_Tag);
var _BackTop = require('./BackTop');var _BackTop2 = _interopRequireDefault(_BackTop);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}exports.default =

{
  Breadcrumb: _Breadcrumb2.default,
  Calendar: _Calendar2.default,
  InputMask: _InputMask2.default,
  InputMoment: _InputMoment2.default,
  InputDate: _InputDate2.default,
  InputTime: _InputTime2.default,
  InputTimeRange: _InputTimeRange2.default,
  InputDateTime: _InputDateTime2.default,
  InputNumeric: _InputNumeric2.default,
  List: _List2.default,
  DatePicker: _DatePicker2.default,
  CollapsePanel: _CollapsePanel2.default,
  LoadingBar: _LoadingBar2.default,
  PropertyList: _PropertyList2.default,
  ComboBox: _ComboBox2.default,
  CheckBox: _Checkbox2.default,
  CheckboxGroup: _Checkbox.CheckboxGroup,
  Radio: _Radio2.default,
  RadioGroup: _Radio.RadioGroup,
  Input: _Input2.default,
  Button: _Button2.default,
  Modal: _Modal2.default,
  Popover: _Popover2.default,
  TextArea: _TextArea2.default,
  AsyncContent: _AsyncContent2.default,
  Progress: _Progress2.default,
  FileUpload: _FileUpload2.default,
  Survey: _Survey2.default,
  ButtonBar: _ButtonBar2.default,
  Tabs: _Tabs2.default,
  Select: _Select2.default,
  Tag: _Tag2.default,
  BackTop: _BackTop2.default };module.exports = exports['default'];