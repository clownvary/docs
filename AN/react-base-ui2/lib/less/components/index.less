@import './grid.less';
@import './icon.less';
@import './backtop.less';
