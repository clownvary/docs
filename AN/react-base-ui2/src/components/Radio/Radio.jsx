import React, { Component } from 'react';
import { bool, number, string, object, func, node, oneOfType } from 'prop-types';
import identity from 'lodash/identity';

import RadioComponent from './RadioComponent';

/** Default PropTypes of Radio.
 * @memberof Radio
*/
const RadioPropTypes = {
  /** Radio value.
   * @type {string|number}
   */
  value: oneOfType([string, number]),
  /** Determines enable/disable state.
   * @type {boolean}
   */
  disabled: bool,
  /** Determines the Radio size.
   * @type {Size}
   */
  size: string,
  /** Field name.
   * @type {string}
   */
  name: string,
  /** Child nodes.
   * @type {node}
   */
  children: node,
  /**  Fires when value change.
   * @event
   * @type {func}
  */
  onChange: func
};

const RadioProps = {
  onChange: identity
};

/** UI component of Radio */
class Radio extends Component {
  static displayName = 'Radio'
  static propTypes = RadioPropTypes;
  static defaultProps = RadioProps;

  static contextTypes = {
    auiRadioGroup: object
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  componentDidMount() {
    Object.defineProperties(this, {
      checked: {
        get() {
          return this.radioComponent.checked;
        },
        set(v) {
          this.radioComponent.checked = !!v;
        }
      }
    });
  }

  setRef = (radioComponent) => {
    this.radioComponent = radioComponent;
  }

  render() {
    const { children, ...rest } = this.props;
    const { auiRadioGroup } = this.context;

    // Proxy `onChange`, `checked` and `disabled` to `auiRadioGroup`
    if (auiRadioGroup) {
      rest.onChange = auiRadioGroup.onChange;
      rest.checked = this.props.value === auiRadioGroup.value;
      rest.disabled = this.props.disabled || auiRadioGroup.disabled;
      rest.size = this.props.size || auiRadioGroup.size;
      rest.name = this.props.name || auiRadioGroup.name;
    }

    return (
      <RadioComponent ref={this.setRef} {...rest}>
        {children}
      </RadioComponent>
    );
  }
}

export default Radio;
