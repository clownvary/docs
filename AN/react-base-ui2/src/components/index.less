@import 'Breadcrumb/index.less';
@import 'Calendar/index.less';
@import 'InputBankCard/index.less';
@import 'InputBase/index.less';
@import 'ColumnList/index.less';
@import 'List/index.less';
@import 'CollapsePanel/index.less';
@import 'Table/index.less';
@import 'InputMoment/index.less';
@import 'InputDateTime/index.less';
@import 'LoadingBar/index.less';
@import 'InputNumeric/index.less';
@import 'PropertyList/index.less';
@import 'Sticky/index.less';
@import 'ComboBox/index.less';
@import 'Dropdown/index.less';
@import 'Dropdown/aaui-dropdown.less';
@import 'Duration/index.less';
@import 'Phone/index.less';
@import 'Alert/index.less';
@import 'DialogBox/index.less';
@import 'AsyncContent/index.less';
@import 'Survey/index.less';
@import 'Survey/vertical.less';
@import 'Tooltip/index.less';
@import 'Progress/index.less';
@import 'FileGallery/index.less';
@import 'MessageBoard/index.less';
@import 'FileUpload/index.less';
@import 'InputTimeRange/index.less';
@import 'ButtonBar/index.less';
@import 'Tabs/index.less';
@import 'Scroller/index.less';
@import 'ResourceCalendar/index.less';
@import 'PCI/index.less';
@import 'Collapse/index.less';
@import 'Steps/index.less';
@import 'SessionCalendar/index.less';
@import 'Select/index.less';
@import 'Tag/index.less';
