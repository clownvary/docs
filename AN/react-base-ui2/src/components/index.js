import Breadcrumb from './Breadcrumb/';
import Calendar from './Calendar/';
import InputMask from './InputMask';
import InputMoment from './InputMoment';
import InputDate from './InputDate';
import InputTime from './InputTime';
import InputTimeRange from './InputTimeRange';
import InputDateTime from './InputDateTime';
import InputNumeric from './InputNumeric';
import List from './List';
import DatePicker from './DatePicker';
import CollapsePanel from './CollapsePanel';
import LoadingBar from './LoadingBar';
import PropertyList from './PropertyList';
import ComboBox from './ComboBox';
import CheckBox, { CheckboxGroup } from './Checkbox';
import Radio, { RadioGroup } from './Radio';
import Input from './Input';
import Button from './Button';
import Modal from './Modal';
import Popover from './Popover';
import TextArea from './TextArea';
import AsyncContent from './AsyncContent';
import Progress from './Progress';
import FileUpload from './FileUpload';
import Survey from './Survey';
import ButtonBar from './ButtonBar';
import Tabs from './Tabs';
import Select from './Select';
import Tag from './Tag';
import BackTop from './BackTop';
import ScreenReaderOnly from './ScreenReaderOnly';

export default {
  Breadcrumb,
  Calendar,
  InputMask,
  InputMoment,
  InputDate,
  InputTime,
  InputTimeRange,
  InputDateTime,
  InputNumeric,
  List,
  DatePicker,
  CollapsePanel,
  LoadingBar,
  PropertyList,
  ComboBox,
  CheckBox,
  CheckboxGroup,
  Radio,
  RadioGroup,
  Input,
  Button,
  Modal,
  Popover,
  TextArea,
  AsyncContent,
  Progress,
  FileUpload,
  Survey,
  ButtonBar,
  Tabs,
  Select,
  Tag,
  BackTop,
  ScreenReaderOnly
};
