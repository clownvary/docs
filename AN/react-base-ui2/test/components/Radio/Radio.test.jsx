import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Radio } from 'src/components/Radio';

describe('components/Radio', () => {
  it('Radio renders fine', () => {
    const snapshot = renderer.create(
      <Radio
        id="snapshot-radio-id"
        size={'md'}
        className={'snapshot-radio-class'}
        toggle
        checked
      >
        snapshot radio
      </Radio>
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('DefaultChecked radio renders fine', () => {
    const snapshot = renderer.create(
      <Radio
        defaultChecked
      />
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Radio component works fine', () => {
    const onRadioChange = jest.fn();
    const radio = mount(<Radio onChange={onRadioChange} />);
    const instance = radio.instance();

    expect(instance.checked).toBeFalsy();

    instance.checked = true;
    expect(instance.checked).toBeTruthy();

    radio.find('input[type="radio"]').simulate('change', { target: { checked: false } });
    radio.setProps({ checked: false });
    expect(instance.checked).toBeFalsy();
    expect(onRadioChange).toHaveBeenCalledTimes(1);
  });

  it('Disabled radio component works fine', () => {
    const onRadioChange = jest.fn();
    const radio = mount(<Radio onChange={onRadioChange} checked />);
    const instance = radio.instance();

    expect(instance.checked).toBeTruthy();

    radio.find('input[type="radio"]').simulate('change', { target: { checked: false } });
    radio.setProps({ checked: false });
    expect(instance.checked).toBeFalsy();
    expect(onRadioChange).toHaveBeenCalledTimes(1);

    radio.setProps({ disabled: true });
    expect(instance.checked).toBeFalsy();
    expect(onRadioChange).toHaveBeenCalledTimes(1);

    radio.find('input[type="radio"]').simulate('change', { target: { checked: true } });
    expect(onRadioChange).toHaveBeenCalledTimes(1);
  });
});
