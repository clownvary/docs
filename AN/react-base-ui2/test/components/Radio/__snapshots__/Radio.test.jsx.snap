// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Radio DefaultChecked radio renders fine 1`] = `
<label
  aria-checked={true}
  aria-disabled={undefined}
  className="radio-wrapper"
  htmlFor="radio_attr_for_1"
  role="radio"
  style={undefined}
>
  <span
    className="radio radio--checked radio--md"
  >
    <input
      checked={true}
      className="radio__input"
      disabled={undefined}
      id="radio_attr_for_1"
      onChange={[Function]}
      type="radio"
    />
    <span
      className="radio__inner"
    />
    <span />
  </span>
</label>
`;

exports[`components/Radio Radio renders fine 1`] = `
<label
  aria-checked={true}
  aria-disabled={undefined}
  className="radio-wrapper snapshot-radio-class"
  htmlFor="snapshot-radio-id"
  role="radio"
  style={undefined}
>
  <span
    className="radio radio--checked radio--md"
  >
    <input
      checked={true}
      className="radio__input"
      disabled={undefined}
      id="snapshot-radio-id"
      onChange={[Function]}
      toggle={true}
      type="radio"
    />
    <span
      className="radio__inner"
    />
    <span>
      snapshot radio
    </span>
  </span>
</label>
`;
